src/components/collapsible.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactNode, FC, useState } from "react"
+import classNames from "classnames"
+
+import ArrowDown from "./icons/arrowDown"
+
+interface Props {
+  /**
+   * False if the element shall be expanded by default
+   * default: true
+   */
+  isInitiallyCollapsed?: boolean
+  /**
+   * A render prop to customize the clickable title of the collapsible element
+   *  - isCollapsed tells you the current state of the collapsible element
+   */
+  renderToggle: (options: { isCollapsed: boolean }) => ReactNode
+  /**
+   * Children to be rendered within the collapse component. The elements will be visible when the component is expanded.
+   */
+  children?: () => ReactNode
+  /**
+   * Function executed when the collapse state has changed.
+   *  - passes the new collapse state as a function parameter
+   */
+  onCollapseStateChange?: (isCollapsed: boolean) => void
```

Comment:
For the sake of brevity, we could also just call it `onChange`. in the context of `Collapsible`, this seems like the only state that can ever change
```suggestion
  onChange?: (isCollapsed: boolean) => void
```

Reply:
In `renderToggle`, we're passing an options object with `isCollapsed` to the callback. shall we keep the two consistent? either always pass a variable or always pass an object?

Reply:
good point. will change it